src/main/java/com/redhat/devtools/lsp4ij/launching/ui/FileTypeServerMappingTablePanel.java

```diff
@@ -0,0 +1,113 @@
+package com.redhat.devtools.lsp4ij.launching.ui;
+
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.fileTypes.FileTypeManager;
+import com.intellij.openapi.ui.ComboBoxTableRenderer;
+import com.intellij.openapi.util.NlsContexts;
+import com.intellij.util.ui.ColumnInfo;
+import com.redhat.devtools.lsp4ij.LanguageServerBundle;
+import com.redhat.devtools.lsp4ij.launching.ServerMappingSettings;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import javax.swing.table.TableCellEditor;
+import javax.swing.table.TableCellRenderer;
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class FileTypeServerMappingTablePanel extends ServerMappingTablePanel {
+
+    public FileTypeServerMappingTablePanel() {
+        this(true);
+    }
+
+    public FileTypeServerMappingTablePanel(boolean editable) {
+        super(new FileTypeColumn(editable), editable);
+        getTable().getEmptyText().setText(LanguageServerBundle.message("new.language.server.dialog.mappings.fileType.no"));
+    }
+
+    @Override
+    protected ServerMappingSettings createServerMappingSettings() {
+        Collection<String> availableFileTypes = getAvailableFileTypes();
+        return ServerMappingSettings.createFileTypeMappingSettings(availableFileTypes.stream().findFirst().get(), null);
+    }
+
+    private static Collection<String> getAvailableFileTypes() {
+        Collection<String> fileTypes = Stream.of(FileTypeManager.getInstance().getRegisteredFileTypes())
+                .map(FileType::getName)
+                .collect(Collectors.toList())
+                .stream().sorted()
+                .collect(Collectors.toList());
+        //val configuredFileTypes = models.tableModel.items.map { it.fileTypeName }.toSet()

```

Comment:
delete that

Reply:
I deleted it, but we need to remove the file types which are already used to avoid filling multiple mappings with the same file type.